site/app/libraries/database/DatabaseQueries.php

```diff
@@ -7401,9 +7401,24 @@ public function deletePoll($poll_id) {
 
     public function getResults($poll_id) {
         $results = [];
-        foreach ($this->getResponses($poll_id) as $option_id => $answer) {
-            $this->course_db->query("SELECT * FROM poll_responses where poll_id = ? and option_id = ?", [$poll_id, $option_id]);
-            $results[$option_id] = count($this->course_db->rows());
+        $this->course_db->query(<<<SQL
+SELECT
+    po.option_id,
+    pr.count
+FROM
+    poll_options AS po
+    LEFT JOIN (
+        SELECT option_id, COUNT(*) AS count FROM poll_responses WHERE poll_id = ? GROUP BY option_id
+    ) AS pr ON pr.option_id = po.option_id
+WHERE
+    po.poll_id = ?
+ORDER BY
+    po.order_id ASC
+SQL,
+            [$poll_id, $poll_id]
+        );
+        foreach ($this->course_db->rows() as $row) {
+            $results[$row["option_id"]] = $row['count'];
```

Comment:
This change isn't necessary to fix the bug, but was just an optimization on the SQL query that existed before, reducing the amount of queries we run from `N+1` where `N` is number of options to `1`.